content/learn/book/the-game-loop/time-and-timers.md

```diff
@@ -0,0 +1,277 @@
++++
+title = "Time and Timers"
+insert_anchor_links = "right"
+[extra]
+weight = 5
++++
+
+Responding to the passage of time is essential for gameplay, animations and audio.
+But doing so correctly is surprisingly subtle.
+
+The [`Time`] resource in Bevy is the source of truth
+for the "current in-game time".
+Unlike [`Instant::now()`], the [`Time`] is fixed for the entire frame,
+rather than reflecting the precise wall clock time at the time of system evaluation.
+This value is set at the start of the frame based on the time supplied by the renderer,
+during the [`TimeSystem`] system set in the [`First`] schedule.
+
+This is helpful for performance reasons, but more critically,
+it ensures consistency of behavior across all of the various bits of game logic.
+
+[`Time`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Time.html
+[`Instant::now()`]: https://doc.rust-lang.org/std/time/struct.Instant.html#method.now
+[`TimeSystem`]: https://docs.rs/bevy/latest/bevy/time/struct.TimeSystem.html
+[`First`]: https://docs.rs/bevy/latest/bevy/app/struct.First.html
+
+## Frame-rate independence and delta time
+
+Suppose we want to move our player to the right:
+
+```rust
+use bevy::prelude::*;
+
+#[derive(Component)]
+struct Player;
+
+fn move_player(mut player_transform: Single<&mut Transform, With<Player>>){
+    const PLAYER_MOVEMENT: f32 = 1.;
+    
+    player_transform.translation.x += PLAYER_MOVEMENT;
+}
+```
+
+Every frame, our player will move 1 unit to the right. Great!
+But what happens when our game stutters, and the frame rate drops?
+Suddenly, rather than moving 60 units per second at our target 60 frames per second,
+our player is moving at an unsteady 20-30 units per second. Oh no!
+
+Instead, we can compensate for this effect by fixing our *speed* (or other rates of change per second),
+and then multiplying by the elapsed time.
+
+```rust, hide_lines=1-4
+# use bevy::prelude::*;
+# 
+# #[derive(Component)]
+# struct Player;
+#
+fn move_player(mut player_transform: Single<&mut Transform, With<Player>>, time: Res<Time>){
+    // At 60 FPS, this will be the same as before
+    const PLAYER_SPEED: f32 = 60.;
+    
+    player_transform.translation.x += PLAYER_SPEED * time.delta_secs_f32();
```

Comment:
`delta_secs_f32` doesn't exist, there's [`delta_secs`](https://dev-docs.bevy.org/bevy/prelude/struct.Time.html#method.delta_secs) and [`delta_secs_f64`](https://dev-docs.bevy.org/bevy/prelude/struct.Time.html#method.delta_secs_f64)

Reply:
```suggestion
    player_transform.translation.x += PLAYER_SPEED * time.delta_secs();
```